lib/deliver/upload_screenshots.rb

```diff
@@ -9,17 +9,45 @@ def upload(options, screenshots)
       v = app.edit_version
       UI.user_error!("Could not find a version to edit for app '#{app.name}'") unless v
 
-      UI.message("Starting with the upload of screenshots...")
+      # md5's of uploaded screenshots
+      checksums_local = {}
+      # md5's of ITC screenshots
+      checksums_remote = {}
+
+      screenshots.each do |screenshot|
+        # checksum of uploaded screenshot
+        md5 = Spaceship::Utilities.get_source_md5(screenshot.path)
+        checksums_local[screenshot.language] ||= {}
+        # to keep order we have to split them per device
+        device_type = Spaceship::Utilities.get_device_name_by_screen(screenshot.screen_size)
+        checksums_local[screenshot.language][device_type] ||= []
+        checksums_local[screenshot.language][device_type] << md5
+      end
 
-      # First, clear all previously uploaded screenshots, but only where we have new ones
-      # screenshots.each do |screenshot|
-      #   to_remove = v.screenshots[screenshot.language].find_all do |current|
-      #     current.device_type == screenshot.device_type
-      #   end
-      #   to_remove.each { |t| t.reset! }
-      # end
+      v.screenshots.each do |lang, screenshots_for_lang|
+        screenshots_for_lang.each do |current|
+          checksum = current.original_file_name.split('_').at(1)
```

Comment:
wouldn't that fail if the screenshot wasn't uploaded by us (e.g. no _ in name) ? I suggest you add tests for this scenario as well.